### DISCUSSION TITLE
[QUESTION] Removing upstream spam flag not working

### DISCUSSION THREAD

**Discussion by Talkabout** [rspamd/rspamd#4440] [General]
Hi guys,

I am using rspamd since years now but I am currently facing an issue I cannot solve.

Some mails I am receiving are getting a spam flag from the upstream server (X-Spam-Flag), but I would like rspamd to ignore this flag and perform its checks without assuming the mail to be spam. What I have tried so far in "milter_headers.conf":

- setting "remove_upstream_spam_flag" => not working
- using "remove-spam-flag" routing => not working
- using "remove-header" routine => not working
- using "remove-headers" routine => not working

For all these cases I have checked the log file for errors but was able to get rid of all of them, still the removal does not work. The resulting headers in the mail are:

![image](https://user-images.githubusercontent.com/10157917/225910760-f75905e8-8c8f-4a9f-a050-84269870f7cd.png)

![image](https://user-images.githubusercontent.com/10157917/225910809-c7c668de-4b03-4e0b-8aad-d26a8a69cb9c.png)

Is the approach of using "milter_headers.conf" correct to make sure the "X-Spam-Flag" is removed PRIOR to the checks rspamd performs? Is there any other way of solving this "problem"?

By the way: to check incoming mails I am using "rspamc", but I think this is correct, right?

Thanks!

**Comment by moisseev** [comment]
The `remove-spam-flag` routine is intended to remove existing spam flag AFTER checks.

If you want Rspamd to ignore existing flags you need to disable `upstream_spam_filters` group or individual symbils.
See [upstream_spam_filters.lua](https://github.com/rspamd/rspamd/blob/master/rules/regexp/upstream_spam_filters.lua) regexp rule.

Please, don't open issues for general questions, use `Discussions` instead.

**Comment by Talkabout** [comment]
I am terribly sorry but I was not sure if this was a question or a bug report :)

Thanks for clarifying!
